internal/xsql/valuer.go

```diff
@@ -777,9 +777,10 @@ func (v *ValuerEval) simpleDataEval(lhs, rhs interface{}, op ast.Token) interfac
 	// Evaluate if both sides are simple types.
 	switch lhs := lhs.(type) {
 	case bool:
-		rhs, ok := rhs.(bool)
+		originRHS := rhs
```

Comment:
Does this problem also happen in other case statement?

Reply:
no